File: java/words/src/test/java/com/example/words/NumericalTest.java
package com.example.words;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

public class NumericalTest {
  @Test
  void testFibonacci() {
    assertEquals(-1, Numerical.fibonacci(-3));
    assertEquals(55, Numerical.fibonacci(10));
    assertEquals(610, Numerical.fibonacci(15));
  }

  @Test
  void testFactorial() {
    assertEquals(-1, Numerical.factorial(-3));
    assertEquals(120, Numerical.factorial(5));
    assertEquals(3628800, Numerical.factorial(10));
  }

  @Test
  void testGcd() {
    assertEquals(6, Numerical.gcd(48, 18));
    assertEquals(25, Numerical.gcd(100, 25));
    assertEquals(1, Numerical.gcd(17, 13));
  }

  @Test
  void testLcm() {
    assertEquals(12, Numerical.lcm(6, 4));
    assertEquals(42, Numerical.lcm(21, 6));
    assertEquals(21, Numerical.lcm(7, 3));
  }

}

File: _experimental/_python/words/test/numerical_test.py
import unittest

import sys
import os
sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))

from src import numerical

class TestFibonacci(unittest.TestCase):
  def test_fibonacci(self):
    self.assertEqual(numerical.fibonacci(-3), -1)
    self.assertEqual(numerical.fibonacci(10), 55)
    self.assertEqual(numerical.fibonacci(15), 610)
  
  def test_factorial(self):
    self.assertEqual(numerical.factorial(-3), -1)
    self.assertEqual(numerical.factorial(5), 120)
    self.assertEqual(numerical.factorial(10), 3628800)

  def test_gcd(self):
    self.assertEqual(numerical.gcd(48, 18), 6)
    self.assertEqual(numerical.gcd(100, 25), 25)
    self.assertEqual(numerical.gcd(17, 13), 1)
  
  def test_lcm(self):
    self.assertEqual(numerical.lcm(6, 4), 12)
    self.assertEqual(numerical.lcm(21, 6), 42)
    self.assertEqual(numerical.lcm(7, 3), 21)